res/controllers/Reloop-Jockey-3-ME-scripts.js

```diff
@@ -0,0 +1,444 @@
+  Jockey3ME.LedShowBeginTimer = engine.beginTimer(500,"Jockey3ME.LedShowBegin()",1); // LedShow Script Starts Here after 500ms
+}
+
+// Sets VuMeter new Value to Leds
+Jockey3ME.fVuMeter = function () {
+  var VuVal1 = engine.getValue("[Channel1]","VuMeter");
+  var VuVal2 = engine.getValue("[Channel2]","VuMeter");
+  var VuVal3 = engine.getValue("[Channel3]","VuMeter");
+  var VuVal4 = engine.getValue("[Channel4]","VuMeter");
+  VuVal1 = VuVal1 * 10;
+  VuVal1 = parseInt(VuVal1);
+  VuVal2 = VuVal2 * 10;
+  VuVal2 = parseInt(VuVal2);
+  VuVal3 = VuVal3 * 10;
+  VuVal3 = parseInt(VuVal3);
+  VuVal4 = VuVal4 * 10;
+  VuVal4 = parseInt(VuVal4);
+  midi.sendShortMsg(0x90,0x21,VuVal1);
+  midi.sendShortMsg(0x91,0x21,VuVal2);
+  midi.sendShortMsg(0x92,0x21,VuVal3);
+  midi.sendShortMsg(0x93,0x21,VuVal4);
+}
+
+Jockey3ME.shutdown = function () {
+  for (var i = 1; i <= 160; i++) {
+    midi.sendShortMsg(0x90,i,0x00);
+    midi.sendShortMsg(0x91,i,0x00);
+    midi.sendShortMsg(0x92,i,0x00);
+    midi.sendShortMsg(0x93,i,0x00);
+  };
+  engine.stopTimer(Jockey3ME.VuMeter);
+  Jockey3ME.VuMeter = 0;
+}
+
+// The button that enables/disables scratching
+Jockey3ME.wheelTouch = function (channel, control, value, status, group) {
+  var currentDeck = parseInt(group.substring(8,9));
+    if (value == 0x7F) {  // Some wheels send 0x90 on press and release, so you need to check the value
+        var alpha = 1.0/8;
+        var beta = alpha/32;
+        engine.scratchEnable(currentDeck, 2048, 45+1/3, alpha, beta);
+    }
+    else {    // If button up
+        engine.scratchDisable(currentDeck);
+    }
+}
+
+// The wheel that actually controls the scratching
+Jockey3ME.wheelTurn = function (channel, control, value, status, group) {
+  var newValue=(value-64);
+  var currentDeck = parseInt(group.substring(8,9));
+    // See if we're scratching. If not, skip this.
+    if (!engine.isScratching(currentDeck)) {
+      if (newValue > 1 || newValue < -1) {
+        newValue /= 2;  // If Jogwheel Resolution is High
+      }
+      engine.setValue(group, "jog", newValue);
+      return;
+   }
+    engine.scratchTick(currentDeck,newValue);
+}
+
+// Hotcues
+Jockey3ME.hotcue_activate = function (channel, control, value, status, group) {
+  Jockey3ME.hotc = Jockey3ME.hotcue_buttonSelect(control);
+
+  if (!Jockey3ME.hotcueClearVal && engine.getValue(group,"hotcue_"+Jockey3ME.hotc+"_enabled") == 0 && value == 0x7F) {
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_activate",1);
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_activate",0);
+  } else if (Jockey3ME.hotcueClearVal && engine.getValue(group,"hotcue_"+Jockey3ME.hotc+"_enabled") == 1 && value == 0x7F) {
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_clear",1);
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_clear",0);
+  } else if (value == 0x7F) {
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_activate",1);
+    engine.setValue(group,"hotcue_"+Jockey3ME.hotc+"_activate",0);
+  }
+}
```

Comment:
This function could be simplified. Just check if the trash button is pressed to clear the hotcue. Otherwise, set hotcue_X_activate. Mixxx takes care of the rest.